### PR TITLE
Create Spring Boot app for Scan On Demand API - GCI 817

### PR DESCRIPTION
**Pull request by YiannisPapastergiou**
Jira story - GCI 817

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplication.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.companieshouse.scanupondemand.orders.api;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class ScanUponDemandApiApplication {
+
+	public static final String APPLICATION_NAMESPACE = "scan-upon-demand.orders.api.ch.gov.uk";
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplication.java:9]: Perhaps just keep the `LoggingUtils` constant?

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplicationTests.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.companieshouse.scanupondemand.orders.api;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplicationTests.java:6]: Is this annotation required here (yet)?

Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplicationTests.java:6]: looks like it is not, it was just copied from the template found here: https://companieshouse.atlassian.net/wiki/spaces/DEV/pages/1295614679/CH+QuickStart+Spring+Boot+App+Template

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplication.java (hunk 1 above)
Comment (YiannisPapastergiou) [reply src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplication.java:9]: I followed the certified-copies approach which is different from other services approaches, which makes it a little bit confusing for me.

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplication.java:9]: Yes, well our approach is evolving over time. We just need one instance of this constant in the app. Observing that in some cases, this constant is used from within LoggingUtils methods, that might be the place for it. But at this point it shouldn't matter which, so long as we only define it once.

Comment (YiannisPapastergiou) [reply src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/ScanUponDemandApiApplication.java:9]: Thanks for the explanation, I had a discussion in parallel with @hwilliams1 and now it clear to me. I will remove it and push it again as well as the other comment for the test class for the sake of good programming practices

File: .gitignore [hunk 3]
```diff
@@ -21,3 +21,13 @@
...
+/target/
+
+# OS files
+.DS_Store
+
+### IntelliJ IDEA ###
+.idea
+*.iws
+*.iml
+*.ipr
```
Comment (rafahop) [inline .gitignore:33]: Should we use our [maven .gitignore template](https://github.com/companieshouse/styleguides/blob/master/gitignore/maven)?

Comment (YiannisPapastergiou) [reply .gitignore]: Check the latest changes please

Comment (rafahop) [reply .gitignore]: I'm not sure whether we need to keep the entries currently in master. Could we just override it with the template?

Comment (YiannisPapastergiou) [reply .gitignore]: done!

**Pull request merged**